flink/src/main/java/org/apache/iceberg/flink/source/split/IcebergSourceSplitStatus.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.flink.source.split;
+
+import java.io.Serializable;
+import javax.annotation.Nullable;
+import org.apache.iceberg.relocated.com.google.common.base.Objects;
+
+public class IcebergSourceSplitStatus implements Serializable {
+
+  public enum Status {
+    UNASSIGNED,
+    ASSIGNED,
+    COMPLETED
+  }
+
+  private final Status status;
+  @Nullable
+  private final Integer assignedSubtaskId;
+
+  /**
+   * The splits are frequently serialized into checkpoints.
+   * Caching the byte representation makes repeated serialization cheap.
```

Comment:
What's the rationale for this optimization? 

I'm not sure if it's going to be expensive to serialize this DS, given that the number of splits the JM will deal with is probably going to be in the 1000s. Even if this was a really high number, there are other easier ways to deal with this, such as only serializing incomplete or un-checkpointed splits.

Reply:
This become irrelevant as `IcebergSourceSplitStatus` is now a simple enum.